llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -126,6 +127,8 @@ namespace {
     TargetSchedModel SchedModel;
     bool PreRegAlloc = false;
     bool HasProfileData = false;
+    Pass *LegacyPass;
```

Comment:
What are these ultimately used for? It's pretty ugly to need to forward these through.

Reply:
`MachineBasicBlock::SplitCriticalEdge` gets the analyses through one that is non-null.

Alternatively we can modify `SplitCriticalEdge` to be passed in the analyses rather than getting it itself.

Reply:
I think most places directly pass in the analyses. This looks like it uses 5 of them, which is kind of an annoyingly large number of them